src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -212,4 +221,28 @@ public void setDataset(Dataset dataset) {
     protected Timestamp getTimestamp() {
         return timestamp;
     }
+
+    protected void checkSystemMetadataKeyIfNeeded(DatasetVersion newVersion, DatasetVersion persistedVersion) throws IllegalCommandException {
+        Set<MetadataBlock> changedMDBs = DatasetVersionDifference.getBlocksWithChanges(newVersion, persistedVersion);
+        changedMDBs.forEach(mdb -> {
+            logger.fine(mdb.getName() + " has been changed");
+        });
+        
+        String smdbString = JvmSettings.METADATA_BLOCK_SYSTEM_METADATA_KEYS.lookupOptional().orElse(null);
+        if (smdbString != null) {
+            JsonObject systemMetadataBlocks = JsonUtil.getJsonObject(smdbString);
+            HttpServletRequest httpServletRequest = getRequest().getHttpServletRequest();
+            if (httpServletRequest != null) {
+                String mdKey = httpServletRequest.getParameter(MetadataBlockServiceBean.SYSTEM_MD_KEY);
+                for (MetadataBlock mdb : changedMDBs) {
+                    if (systemMetadataBlocks.containsKey(mdb.getName())) {
```

Comment:
A thought: it has advantages to use a JSON object here, yes. What would also work: create a `JvmSetting` with a placeholder at the end. Less complexity, no costly JSON parsing.

Asking for `dataverse.metadata.block-system-metadata-keys.citation` would return the key `String` if set for a certain block. You can access this variable setting via the same `JvmSettings.lookupOptional()`, but you need to pass the name of the block as the var arg.